src/main/java/org/opensearch/security/state/SecurityClusterNonManagerListener.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ */
+package org.opensearch.security.state;
+
+import org.opensearch.cluster.ClusterChangedEvent;
+import org.opensearch.cluster.ClusterStateListener;
+import org.opensearch.security.configuration.ConfigurationRepository;
+import org.opensearch.security.securityconf.impl.CType;
+
+public class SecurityClusterNonManagerListener implements ClusterStateListener {
+
+    private final ConfigurationRepository configurationRepository;
+
+    public SecurityClusterNonManagerListener(final ConfigurationRepository configurationRepository) {
+        this.configurationRepository = configurationRepository;
+    }
+
+    @Override
+    public void clusterChanged(final ClusterChangedEvent clusterChangedEvent) {
+        if (clusterChangedEvent.localNodeClusterManager()) {
+            return;
+        }
+        final var minNodeVersion = clusterChangedEvent.state().nodes().getMinNodeVersion();
+        final var maxNodeVersion = clusterChangedEvent.state().nodes().getMaxNodeVersion();
+        if (clusterChangedEvent.state().custom(SecurityMetadata.TYPE, SecurityMetadata.DEFAULT).isSecurityApplied()) {
```

Comment:
Will this be triggered on any cluster change event, or only changes to this key?  Seems like there should be some filtering before we reload the configuration.

Reply:
it triggers on each cluster state change. It is how OS works with the state.